pkg/storegateway/series_chunks.go

```diff
@@ -345,7 +357,7 @@ func (c *loadingSeriesChunksSetIterator) Next() (retHasNext bool) {
 	}
 
 	// Create a batched memory pool that can be released all at once.
-	chunksPool := &pool.BatchBytes{Delegate: c.chunksPool}
+	chunksPool := pool.NewSafeSlabPool[byte](chunkBytesSlicePool, chunkBytesSlabSize)
```

Comment:
Please remove `loadingSeriesChunksSetIterator.chunksPool` and do not pass it anymore to `newLoadingSeriesChunksSetIterator()` because it's not used anymore. Now we use the global one. This is the reason why the test `TestLoadingSeriesChunksSetIterator` is failing: we're asserting on a pool we're not using. Instead look at the comment I left on `chunkBytesSlicePool`.